airbyte-oauth/src/main/java/io/airbyte/oauth/google/GoogleOAuthFlow.java

```diff
@@ -53,59 +55,53 @@
 
   private final HttpClient httpClient;
 
-  private static final String GOOGLE_ANALYTICS_CONSENT_URL = "https://accounts.google.com/o/oauth2/v2/auth";
-  private static final String GOOGLE_ANALYTICS_ACCESS_TOKEN_URL = "https://oauth2.googleapis.com/token";
-  @VisibleForTesting
-  static final String GOOGLE_ANALYTICS_SCOPE = "https%3A//www.googleapis.com/auth/analytics.readonly";
-  private static final List<String> GOOGLE_QUERY_PARAMETERS = List.of(
-      String.format("scope=%s", GOOGLE_ANALYTICS_SCOPE),
-      "access_type=offline",
-      "include_granted_scopes=true",
-      "response_type=code",
-      "prompt=consent");
+  private final String consentUrl = "https://accounts.google.com/o/oauth2/v2/auth";
+  private final String accessTokenUrl = "https://oauth2.googleapis.com/token";
+
+  private final String scope;
+  private final List<String> googleQueryParameters;
 
   private final ConfigRepository configRepository;
 
-  public GoogleOAuthFlow(ConfigRepository configRepository) {
-    this(configRepository, HttpClient.newBuilder().version(Version.HTTP_1_1).build());
+  public GoogleOAuthFlow(ConfigRepository configRepository, String scope) {
+    this(configRepository, scope, HttpClient.newBuilder().version(Version.HTTP_1_1).build());
   }
 
   @VisibleForTesting
-  GoogleOAuthFlow(ConfigRepository configRepository, HttpClient httpClient) {
+  GoogleOAuthFlow(ConfigRepository configRepository, String scope, HttpClient httpClient) {
     this.configRepository = configRepository;
     this.httpClient = httpClient;
+    this.scope = scope;
+    this.googleQueryParameters = List.of(
+        String.format("scope=%s", this.scope),
```

Comment:
See my comments on the encoding of scope parameters in the child classes.

This should probably look like:
```suggestion
        String.format("scope=%s", URLEncoder.encode(this.scope, StandardCharsets.UTF_8.toString())),
```